internal/database/repos.go

```diff
@@ -910,8 +910,11 @@ func (s *repoStore) listSQL(ctx context.Context, opt ReposListOptions) (*sqlf.Qu
 	if opt.OnlyArchived {
 		where = append(where, sqlf.Sprintf("archived"))
 	}
+	// Since https://github.com/sourcegraph/sourcegraph/pull/35633 there is no need to do an anti-join
+	// with gitserver_repos table (checking for such repos that are present in repo but absent in gitserver_repos
+	// table) because repo table is strictly consistent with gitserver_repos table.
 	if opt.NoCloned {
-		where = append(where, sqlf.Sprintf("(gr.clone_status = 'not_cloned' OR gr.clone_status IS NULL)"))
+		where = append(where, sqlf.Sprintf("(gr.clone_status = ANY(ARRAY['not_cloned', 'cloning']))"))
```

Comment:
nit: I find this more readable

```suggestion
		where = append(where, sqlf.Sprintf("(gr.clone_status in ('not_cloned', 'cloning'))"))
```